core/types/snapshot_nodes.go

```diff
@@ -809,6 +825,14 @@ func PayloadFromProto(p *snapshot.Payload) *Payload {
 		ret.Data = PayloadLiquiditySuppliedFromProto(dt)
 	case *snapshot.Payload_LiquidityTarget:
 		ret.Data = PayloadLiquidityTargetFromProto(dt)
+	case *snapshot.Payload_LiquidityV2Provisions:
```

Comment:
shouldn't there be a similar bit for the intoProto?

Reply:
Nice catch.